### DISCUSSION TITLE
Help with writing a query

### DISCUSSION THREAD

**Discussion by simonmandlik** [nvim-treesitter/nvim-treesitter#5432] [Q&A] [answered]
Hi,

I'm working on a `julia` support for [nvim-treesitter-endwise](https://github.com/RRethy/nvim-treesitter-endwise) and encountered the following problem.

How can I write a matching TS query for something like this?

Code:
```julia
module Foo<cursor>
```

`:InspectTree`:
```
"module" ; [1:1 - 6] julia
(identifier) ; [1:8 - 10] julia
"\n" ; [1:11 - 2:0] julia
```

The following queries successfully match the individual nodes:
```
("module") @test
(identifier) @test
```

But I can't match both of them. None of the following work in the editor:
```
("module" (identifier)) @test
(("module" (identifier))) @test
(("module") (identifier)) @test
```

**Comment by amaanq** [comment]
check the top level node, it might not be visible in InspectTree but should be with tree-sitter cli or just checking the first rule of the grammar, Julia's seems to be source_file

```scm
(source_file
  "module" @test
  (identifier) @test)
```
 
I don't know Julia enough to say if that's a valid query or if your code is valid, but going off what I see here this is my best guess

**Reply by simonmandlik** [reply, marked as answer]
When parsing with cli, I got completely different output:

```
(ERROR [0, 0] - [1, 0]
  (identifier [0, 7] - [0, 10]))
test.jl 0 ms    (ERROR [0, 0] - [1, 0])
```

and, indeed, it works

```
((ERROR "module" @a (identifier) @b) @c)
```

I had no idea `:InspectTree` would show such a different output! 😮

**Comment by clason** [comment]
Tangential, but while you're asking: be aware that the whole module system will be removed in the near future, so nvim-treesitter-endwise should switch to a standalone module sooner rather than later (check nvim-treesitter-context for an example).

**Reply by simonmandlik** [reply]
Yes, there is already an issue for that https://github.com/RRethy/nvim-treesitter-endwise/issues/27
